pkg/controller/installation/typha_autoscaler.go

```diff
@@ -65,10 +70,14 @@ func typhaAutoscalerPeriod(syncPeriod time.Duration) typhaAutoscalerOption {
 
 // newTyphaAutoscaler creates a new Typha autoscaler, optionally applying any options to the default autoscaler instance.
 // The default sync period is 2 minutes.
-func newTyphaAutoscaler(client client.Client, options ...typhaAutoscalerOption) *typhaAutoscaler {
-	ta := new(typhaAutoscaler)
-	ta.client = client
-	ta.syncPeriod = defaultTyphaAutoscalerSyncPeriod
+func newTyphaAutoscaler(client client.Client, statusManager status.StatusManager, options ...typhaAutoscalerOption) *typhaAutoscaler {
+	ta := &typhaAutoscaler{
+		client:         client,
+		statusManager:  statusManager,
+		syncPeriod:     defaultTyphaAutoscalerSyncPeriod,
+		triggerRunChan: make(chan chan error, 100),
```

Comment:
Should we really be making these channels with 100 buffer? It should be fine to be 1 for both.

Reply:
Meh, probably not. I guess we don't need to, the go routine is essentially locked while triggerRun and isDegraded are being called anyway.

Reply:
I set it to 0, which means it blocks until the object is read off of the channel. I think that's probably what we want.